### PR TITLE
Add condition in template about user.current_location

### PR DESCRIPTION
**Pull request by Ydrasil**

### CODE REVIEW

File: blog/test.py [hunk 1]
```diff
@@ -11,11 +11,23 @@
 from trips import views
 
 
+def create_user():
+    user = User(email='emailtest@test.com',
```
Comment (vinyll) [inline blog/test.py]: You could move this into the `setUp` of the `ArticleTest` class:
`self.user = User(…`

File: blog/test.py [hunk 2]
```diff
@@ -11,11 +11,23 @@
...
+                username='emailtest').set_password('testtest').save()
+    return user
+
+
+def login_user(self):
```
Comment (vinyll) [inline blog/test.py]: Move this method into the `ArticleTest` class

File: blog/test.py [hunk 3]
```diff
@@ -11,11 +11,23 @@
...
+    data = {
+        'email': 'emailtest@test.com',
+        'password': 'testtest',
+    }
+    self.client.post('/login', data=data, follow_redirects=True)
```
Comment (vinyll) [inline blog/test.py]: Add a trailing slash

File: blog/test.py [hunk 4]
```diff
@@ -27,59 +39,46 @@ def test_articles_empty(self):
...
         result = self.client.get('/articles')
         self.assertIn(b'<article class=preview>', result.data)
 
     def test_read_article_unauthenticated(self):
-        article = Article(title='<h1>title</h1>', content='<p>content</p>')
-        article.author = self.user
-        article.save()
+        article = Article(title='title',
+                          content='<p>content</p>',
+                          author=create_user()).save()
```
Comment (vinyll) [inline blog/test.py]: `author=self.user`

File: blog/test.py [hunk 5]
```diff
@@ -27,59 +39,46 @@ def test_articles_empty(self):
...
-        data = {
-            'email': 'emailtest@test.com',
-            'password': 'testtest',
-        }
-        self.client.post('/login', data=data, follow_redirects=True)
-        article = Article(title='title', content='<p>content</p>')
-        article.author = self.user
-        article.save()
+        user = create_user()
+        login_user(self)
```
Comment (vinyll) [inline blog/test.py]: `login_user(self)`? Why passing `self`??

Comment (vinyll) [reply blog/test.py]: If you move it into a method of the class it will be invalid anyways.

File: blog/test.py [hunk 6]
```diff
@@ -27,59 +39,46 @@ def test_articles_empty(self):
...
 
     def test_article_write_get(self):
-        data = {
-            'email': 'emailtest@test.com',
-            'password': 'testtest',
-        }
-        self.client.post('/login', data=data, follow_redirects=True)
+        create_user()
+        login_user(self)
         result = self.client.get('/article/write')
```
Comment (vinyll) [inline blog/test.py]: trailing slash

File: blog/test.py [hunk 7]
```diff
@@ -27,59 +39,46 @@ def test_articles_empty(self):
...
-        self.client.post('/login', data=data, follow_redirects=True)
-        article = Article(title='title', content='<p>content</p>')
-        article.author = self.user
-        article.save()
+        user = create_user()
+        login_user(self)
+        article = Article(title='title',
+                          content='<p>content</p>',
+                          author=user).save()
         result = self.client.get('/@emailtest/{}-{}'.format(article.slug,
```
Comment (vinyll) [inline blog/test.py]: trailing slash

**Review by vinyll — CHANGES_REQUESTED** [review]
You may need to `rebased`; I see the trailing slashes are not considered here. Or did we not merge it?

**Comment by vinyll** [comment]
@Ydrasil: Can you give here the reason why this is closed?

**Comment by Ydrasil** [comment]
I've reopen an other PR for this #159 
And this PR is for master instead of privacy-location-names

**Pull request closed without merging**